File: 4_gestion_de_archivos/leer.py
import os

# Definir la ruta del archivo
file_path = "clientes_cuentas.txt"

# Intentar abrir el archivo
try:
    with open(file_path, 'r') as input_file:
        # Imprimir el encabezado con formato
        print(f"{'Account':<10}{'Name':<13}{'Balance':>7}")

        # Leer cada registro del archivo y mostrarlo
        for line in input_file:
            parts = line.split()
            if len(parts) == 3:
                account = parts[0]
                name = parts[1]
                balance = float(parts[2])
                
                # Imprimir cada registro con el formato especificado
                print(f"{account:<10}{name:<13}{balance:>7.2f}")
except FileNotFoundError:
    print("No se pudo abrir el archivo")
    exit(1)

File: 4_gestion_de_archivos/leer.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <iomanip> // Para usar std::setw, std::left, std::right, std::fixed, std::setprecision

using namespace std;

int main() {
    // Abrir el archivo de entrada usando ifstream
    ifstream input{"clientes_cuentas.txt", ios::in};

    // Verificar si el archivo se abrió correctamente
    if (input) {
        // Imprimir el encabezado con formato
        cout << left << setw(10) << "Account"
             << setw(13) << "Name"
             << right << setw(7) << "Balance" << "\n";

        int account;
        string name;
        double balance;

        // Mostrar cada registro en el archivo
        while (input >> account >> name >> balance) {
            cout << left << setw(10) << account
                 << setw(13) << name
                 << right << setw(7) << fixed << setprecision(2) << balance << "\n";
        }
    } else {
        // Mensaje de error si no se puede abrir el archivo
        cerr << "No se pudo abrir el archivo\n";
        exit(EXIT_FAILURE);
    }

    return 0;
}
